packages/types/src/options.ts

```diff
@@ -5,9 +5,9 @@ import { CaptureContext } from './scope';
 import { SdkMetadata } from './sdkmetadata';
 import { StackLineParser, StackParser } from './stacktrace';
 import { SamplingContext } from './transaction';
-import { Transport, TransportClass, TransportOptions } from './transport';
+import { BaseTransportOptions, NewTransport } from './transport';
 
-export interface ClientOptions {
+export interface ClientOptions<TO extends BaseTransportOptions = BaseTransportOptions> {
```

Comment:
This is now generic to a set of transport options. In a follow up PR, I'm going to insert in the `BrowserTransportOptions` and `NodeTransportOptions` accordingly.

Reply:
I think that's a good idea!